guides/hack/15-asynchronous-operations/19-extensions.md

```diff
@@ -3,24 +3,63 @@ infrastructure. Async is especially useful with database access and caching, web
 
 ## MySQL
 
-The async MySQL extension is similar to the [`mysqli`](http://php.net/manual/en/book.mysqli.php) extension that comes with HHVM. The async
-MySQL extension is primarily used for asynchronously creating connections and querying MySQL databases.
+The async MySQL extension is a built-in way to execute queries against databases while running Hack code when the database is processing.
+The async MySQL extension is primarily used for asynchronously creating connections and querying MySQL databases.
 
 The [full API](../reference/class/AsyncMysqlConnection/) contains all of the classes and methods available for accessing MySQL via async; we
 will only cover a few of the more common scenarios here.
 
 The primary class for connecting to a MySQL database is [`AsyncMysqlConnectionPool`](../reference/class/AsyncMysqlClient/) and its
 primary method is the `async` [`connect`](../reference/class/AsyncMysqlClient/connect/).
 
-The primary class for querying a database is [`AsyncMysqlConnection`](../reference/class/AsyncMysqlConnection/) with the two main
-query methods, [`query`](../reference/class/AsyncMysqlConnection/query/) and [`queryf`](../reference/class/AsyncMysqlConnection/queryf/),
-both of which are `async`. There is also a function to ensure that queries to be executed are safe called
+The primary class for querying a database is [`AsyncMysqlConnection`](../reference/class/AsyncMysqlConnection/) with the three main query methods:
+[`query`](../reference/class/AsyncMysqlConnection/query/),
+[`queryf`](../reference/class/AsyncMysqlConnection/queryf/), and
+[`queryAsync`](../references/class/AsyncMysqlConnection/queryAsync)
+all of which are `async`. The naming conventions for async method have not been applied consistently.
+There is also a function to ensure that queries to be executed are safe called
 [`escapeString`](../reference/class/AsyncMysqlConnection/escapeString/).
 
+## Choosing a query method
+
+### [`queryAsync`](../references/class/AsyncMysqlConnection/queryAsync)
```

Comment:
From test:

```
1) HHVM\UserDocumentation\Tests\InternalLinksTest::testInternalLink with data set #2037
1084>>> 404: /hack/references/class/AsyncMysqlConnection/queryAsync
1085>>> Linked from:
1086>>>  - /hack/asynchronous-operations/extensions
```

- need a trailing /
- need `reference`, singular
- prefer `/hack/reference` rather than relative path throughout
- ... but prefer just `AsyncMysqlConnection::queryAsync` in backticks over that - will automatically be converted to a link. This would be a good change for most of this file, but doesn't need to all be fixed here